layer/src/conv.rs

```diff
@@ -182,13 +254,15 @@ mod tests {
     use ndarray::arr3;
 
     use super::*;
+    use crate::activation::Linear;
     use test_utils::assert_approx_eq;
 
     #[test]
     fn test_conv1d_nonsingleton_kernel() {
         let weights = Array1::range(0., 18., 1.).into_shape((2, 3, 3)).unwrap();
+        let bias = Array1::zeros(2);
         let input = Array1::range(0., 30., 1.).into_shape((2, 3, 5)).unwrap();
-        let output = Conv1D::new(weights, None, 1, 0, 1, 1)
+        let output = Conv1D::new(weights, bias, Linear, 1, 0, 1, 1)
```

Comment:
adding a test for an activation function other than the identity function is not changing the result of the conv layer (pre-activation), but would just test this activation function. those kind of tests should be unit tests in the activation function module.

btw, maybe we should also rename `Linear` to `Identity` to make clear what this activation function is precisely about (in another pr).

Reply:
it would test that the activation function is actually called.

Reply:
although i'd agree to some extend, it can easily be confirmed that the function call exists by looking at the code. checking if the function is actually called then boils down to checking if the code is compiled correctly as written.